docs/reference/snapshot-restore/apis/restore-snapshot-api.asciidoc

```diff
@@ -0,0 +1,230 @@
+[[restore-snapshot-api]]
+=== Restore snapshot API
+++++
+<titleabbrev>Restore snapshot</titleabbrev>
+++++
+
+Restores a <<snapshot-restore,snapshot>> of a cluster or specified data streams and indices.
+
+////
+[source,console]
+----
+PUT /_snapshot/my_repository
+{
+  "type": "fs",
+  "settings": {
+    "location": "my_backup_location"
+  }
+}
+
+PUT /_snapshot/my_repository/my_snapshot?wait_for_completion=true
+
+PUT /index_1
+
+PUT /index_2
+
+PUT /index_3
+
+PUT /index_4
+
+PUT _snapshot/my_repository/snapshot_2?wait_for_completion=true
+{
+  "indices": "index_3,index_4",
+  "ignore_unavailable": true,
+  "include_global_state": false,
+  "metadata": {
+    "taken_by": "Elastic Machine",
+    "taken_because": "backup testing"
+  }
+}
+
+POST /index_1/_close
+
+POST /index_2/_close
+
+POST /index_3/_close
+
+POST /index_4/_close
+
+----
+// TESTSETUP
+////
+
+[source,console]
+----
+POST /_snapshot/my_repository/my_snapshot/_restore
+----
+// TEST[s/_restore/_restore?wait_for_completion=true/]
+
+[[restore-snapshot-api-request]]
+==== {api-request-title}
+
+`POST /_snapshot/<repository>/<snapshot>`
+
+[[restore-snapshot-api-desc]]
+==== {api-description-title}
+
+Use the restore snapshot API to restore a snapshot of a cluster, including all data streams and indices in the snapshot. If you do not want to restore the entire snapshot, you can select specific data streams or indices to restore.
+
+NOTE: You cannot restore a data stream if a stream with the same name already
+exists.
+
+You can run the restore operation only on a functioning cluster. Existing indices can only be restored if they are <<indices-close,closed>> and have the same number of shards as the indices in the snapshot. The restore operation automatically opens restored indices if they were closed and creates new indices if they do not exist in the cluster.
+
+If a data stream is restored, its backing indices are also restored. However, restored backing indices are not automatically
+added to any existing data streams. For example, if only the
+`.ds-logs-000003` backing index is restored from a snapshot, it is not automatically added to the existing `logs` data stream.
```

Comment:
This paragraph is a bit confusing and seems to conflate two ideas.

You can restore an entire data stream. This also restores the stream's backing indices.

Separately, you can also restore _individual_ backing indices without restoring an entire data stream. However, these restored backing indices are treated as regular indices. They aren't auto-added to any existing data stream.

Reply:
Thanks for clarifying this distinction @jrodewig. I think that this rendition explains the difference better:

> If a data stream is restored, its backing indices are also restored. Alternatively, you can restore individual backing indices without restoring an entire data stream. If you restore individual backing indices, they are not automatically added to any existing data stream. For example, if only the `.ds-logs-00003` backing index is restored from a snapshot, it is not automatically added to the existing `logs` data stream.